test/units/modules/test_iptables.py

```diff
@@ -1,1174 +1,1402 @@
+        }
+    )
+    commands_results = [
+        (0, "", ""),
+    ]
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "mangle",
+        "-C",
+        "PREROUTING",
+        "-p",
+        "udp",
+        "-d",
+        "127.0.0.1",
+        "-m",
+        "state",
+        "-j",
+        "TEE",
+        "--gateway",
+        "192.168.10.1",
+        "-i",
+        "eth0",
+        "--destination-port",
+        "9521",
+        "--state",
+        "NEW",
+    ]
+
+
+@pytest.mark.parametrize(
+    ("test_input"),
+    [
+        pytest.param('flags=ALL flags_set="ACK,RST,SYN,FIN"', id="tcp_flags_str"),
+        pytest.param({"flags": "ALL", "flags_set": "ACK,RST,SYN,FIN"}, id="tcp_flags_dict"),
+        pytest.param({"flags": ["ALL"], "flags_set": ["ACK", "RST", "SYN", "FIN"]}, id="tcp_flags_list"),
+    ],
+)
+def test_tcp_flags(mocker, mock_basic_commands, test_input):
+    """Test various ways of inputting tcp_flags."""
+    rule_data = {
+        "chain": "OUTPUT",
+        "protocol": "tcp",
+        "jump": "DROP",
+        "tcp_flags": test_input,
+    }
+
+    set_module_args(rule_data)
+
+    commands_results = [
+        (0, "", ""),
+    ]
+
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "filter",
+        "-C",
+        "OUTPUT",
+        "-p",
+        "tcp",
+        "--tcp-flags",
+        "ALL",
+        "ACK,RST,SYN,FIN",
+        "-j",
+        "DROP",
+    ]
+
+
+@pytest.mark.parametrize(
+    ("log_level"),
+    [
+        pytest.param("0", id="log_level_0"),
+        pytest.param("1", id="log_level_1"),
+        pytest.param("2", id="log_level_2"),
+        pytest.param("3", id="log_level_3"),
+        pytest.param("4", id="log_level_4"),
+        pytest.param("5", id="log_level_5"),
+        pytest.param("6", id="log_level_6"),
+        pytest.param("7", id="log_level_7"),
+        pytest.param("emerg", id="log_level_emerg"),
+        pytest.param("alert", id="log_level_alert"),
+        pytest.param("crit", id="log_level_crit"),
+        pytest.param("error", id="log_level_error"),
+        pytest.param("warning", id="log_level_warning"),
+        pytest.param("notice", id="log_level_notice"),
+        pytest.param("info", id="log_level_info"),
+        pytest.param("debug", id="log_level_debug"),
+    ],
+)
```

Comment:
I think this is one case that could use an `ids` callable instead of using `pytest.param()`
```suggestion
    ids=lambda param: f"log-level-{param !s}",
)
```

OTOH, maybe even this is unnecessary, because `log_level` is already a test function name so identifiers like `test_log_level[debug]` are pretty straightforward to distinguish in the test report so there's no need to repeat the same thing in every param. If anything `test_log_level[log_level_debug]` feels rather redundant compared to the default.